### DISCUSSION TITLE
DNS是否可以增加对ipset的读取

### DISCUSSION THREAD

**Discussion by om2mo** [shadowsocks/shadowsocks-rust#525] [Ideas]
内存读取效率高，iptables也是使用ipset，这样免去了维护ACL和ipset两份列表的麻烦

**Comment by zonyitoo** [comment]
1. ACL also stores all IP networks in a data structure **in memory**. So there would be no significant performance difference between ACL and `ipset`.
2. Does `ipset` have C APIs? Maybe no. The only [`libipset`](https://ipset.netfilter.org/libipset.man.html) doesn't seem to provide an API for `test`ing whether an IP was in an existed `ipset`.

**Reply by LuoZijun** [reply]
我打赌这个没有 `iana-ip-db` 里面的 `acl` 快。

**Reply by metercai** [reply]
> 2. Does `ipset` have C APIs? Maybe no. The only [`libipset`](https://ipset.netfilter.org/libipset.man.html) doesn't seem to provide an API for `test`ing whether an IP was in an existed `ipset`.
```
ipset test SETNAME TEST-ENTRY [ TEST-OPTIONS ]
```
